### PR TITLE
Chore: Update authlib version

### PR DESCRIPTION
**Pull request by alexanderzobnin**
**What is this feature?**

Upate authz library to latest version.

**Why do we need this feature?**

We need latest changes to start aligning clients to the new interface.

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/services/authz/client.go [hunk 1]
```diff
@@ -103,15 +102,13 @@ func newInProcLegacyClient(server *legacyServer) (authzlib.MultiTenantClient, er
 	)
 
 	return authzlib.NewLegacyClient(
-		&authzlib.MultiTenantClientConfig{},
+		&authzlib.ClientConfig{},
 		authzlib.WithGrpcConnectionLCOption(channel),
-		// nolint:staticcheck
-		authzlib.WithNamespaceFormatterLCOption(claims.OrgNamespaceFormatter),
```
Comment (alexanderzobnin) [inline pkg/services/authz/client.go:109]: I haven't found a corresponding function so I assume it's not used anymore.

File: pkg/services/authz/server.go [hunk 2]
```diff
@@ -51,14 +53,18 @@ func (s *legacyServer) Read(ctx context.Context, req *authzv1.ReadRequest) (*aut
 
 	action := req.GetAction()
 	subject := req.GetSubject()
-	stackID := req.GetStackId() // TODO can we consider the stackID as the orgID?
+	namespace := req.GetNamespace() // TODO can we consider the stackID as the orgID?
```
Comment (alexanderzobnin) [inline pkg/services/authz/server.go:56]: This one I'm not sure. What's a format now? Do we have a special helper function to extract orgId from name space?

Comment (gamab) [reply pkg/services/authz/server.go:56]: You can use the authlib claims to parse this:
```suggestion
	namespace := req.GetNamespace() 
	
	info, err := claims.ParseNamespace(namespace)
	if err != nil || info.OrgID == 0 {
		return "", fmt.Errorf("invalid namespace: %s", namespace)
	}
```

File: pkg/services/authz/client.go (hunk 1 above)
Comment (gamab) [reply pkg/services/authz/client.go:109]: Yes, it's actually another interceptor.

**Review by drclau — APPROVED** [review]
Other than Gabriel's suggestions, LGTM 🚀

**Review by gamab — APPROVED** [review]

**Pull request merged**